### PR TITLE
Add GitHub Actions workflow

### LINKED ISSUE #4: Add GitHub Actions workflow
### Description

Add a GitHub Actions workflow to perform automated validations for new PRs.

### Acceptance Criteria

- The newly added GitHub Actions workflow is triggered with every new PR
- The following validations are carried out:
  - Validate if the necessary config files are present in the root directory (app.json and native token.json)
  - Validate to ensure that no additional JS files are present in the submission unless whitelisted
  - Validate the submission against the established schema
  - Validate the chain information against on-chain? E.g Check the `chainID` of the serviceURL against the on-chain data by querying the on-chain endpoint

### PR DESCRIPTION
**Pull request by vardan10**
### What was the problem?
This PR resolves #4

### How was it solved?

- [x] Moved validation tool to root dir.
- [x] Added validation to check if grandparent directory of `app.json` and `nativetoken.json` is a network dir.
- [x] Added validation to check of `chainName` present in `app.json` corresponds to parent dir of `app.json`.
- [x] Added GitHub actions workflow.
- [x] Modify validation tool to accept files changed in PR.
- [x] Update App and token URLs

### How was it tested?

- [x] Locally using [act](https://github.com/nektos/act)

### CODE REVIEW

File: validation/src/index.js [hunk 1]
```diff
@@ -12,33 +12,43 @@
...
-	// Get all app.json and nativetokens.json files
-	const files = [];
-	for (let i = 0; i < networkDirs.length; i++) {
-		/* eslint-disable no-await-in-loop */
-		files.push(...await getNestedFilesByName(networkDirs[i], Object.values(config.filename)));
-		/* eslint-enable no-await-in-loop */
+	const allNetworkDirs = await getNetworkDirs(config.rootDir);
+
+	// Get all modified files
+	const filesChanged = process.argv.slice(2);
```
[the remark this answers is not in the document]
Comment (vardan10) [reply validation/src/index.js]: Added a check to validate if user has added/updated any non-network directory

File: alphanet/Colecti/nativetokens.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
             "displayDenom": "col",
             "symbol": "COL",
             "logo": {
-                "png": "https://lisk-qa.ams3.digitaloceanspaces.com/colecti.png",
+                "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/alphanet/Colecti/images/tokens/lisk.png",
```
Comment (sameersubudhi) [inline alphanet/Colecti/nativetokens.json:24]:
```suggestion
                "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/alphanet/Colecti/images/tokens/colecti.png",
```

File: alphanet/Enevti/app.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "genesisURL": "",
     "projectPage": "https://enevti.com",
     "logo": {
-        "png": "https://media.lisk.com/init/enevti_logo_0dbeff61b1.png",
+        "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/alphanet/Enevti/images/application/lisk.png",
```
Comment (sameersubudhi) [inline alphanet/Enevti/app.json:10]:
```suggestion
        "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/alphanet/Enevti/images/application/enevti.png",
```

File: alphanet/Enevti/nativetokens.json [hunk 4]
```diff
@@ -21,7 +21,7 @@
             "displayDenom": "envt",
             "symbol": "ENVT",
             "logo": {
-                "png": "https://media.lisk.com/init/enevti_logo_0dbeff61b1.png",
+                "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/alphanet/Enevti/images/tokens/lisk.png",
```
Comment (sameersubudhi) [inline alphanet/Enevti/nativetokens.json:24]:
```suggestion
                "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/alphanet/Enevti/images/tokens/enevti.png",
```

File: devnet/Colecti/nativetokens.json [hunk 5]
```diff
@@ -21,7 +21,7 @@
             "displayDenom": "col",
             "symbol": "COL",
             "logo": {
-                "png": "https://lisk-qa.ams3.digitaloceanspaces.com/colecti.png",
+                "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/devnet/Colecti/images/tokens/lisk.png",
```
Comment (sameersubudhi) [inline devnet/Colecti/nativetokens.json:24]:
```suggestion
                "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/devnet/Colecti/images/tokens/colecti.png",
```

File: devnet/Enevti/app.json [hunk 6]
```diff
@@ -7,7 +7,7 @@
     "genesisURL": "",
     "projectPage": "https://enevti.com",
     "logo": {
-        "png": "https://media.lisk.com/init/enevti_logo_0dbeff61b1.png",
+        "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/devnet/Enevti/images/application/lisk.png",
```
Comment (sameersubudhi) [inline devnet/Enevti/app.json:10]:
```suggestion
        "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/devnet/Enevti/images/application/enevti.png",
```

File: devnet/Enevti/nativetokens.json [hunk 7]
```diff
@@ -21,7 +21,7 @@
             "displayDenom": "envt",
             "symbol": "ENVT",
             "logo": {
-                "png": "https://media.lisk.com/init/enevti_logo_0dbeff61b1.png",
+                "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/devnet/Enevti/images/tokens/lisk.png",
```
Comment (sameersubudhi) [inline devnet/Enevti/nativetokens.json:24]:
```suggestion
                "png": "https://raw.githubusercontent.com/LiskHQ/app-registry/main/devnet/Enevti/images/tokens/enevti.png",
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: validation/src/index.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+	const allChangedFiles = process.argv.slice(2);
+
+	// Get all app dir added or modified inside network dirs
+	const changedAppDirs = new Set();
+	for (let i = 0; i < allChangedFiles.length; i++) {
+		const dir = allChangedFiles[i].split('/').slice(0, 2).join('/');
+		if (dir.trim() && config.knownNetworks.includes(dir.split('/')[0])) {
+			changedAppDirs.add(path.resolve(dir));
+		} else {
+			throw new Error('Added/Updated a file not belonging to network directory.');
```
Comment (priojeetpriyom) [inline validation/src/index.js:34]: printing the filepath will help with debugging

```suggestion
			throw new Error(`Added/Updated a file not belonging to network directory. filepath:${allChangedFiles[i]}`);
```

File: validation/src/index.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+	const allChangedFiles = process.argv.slice(2);
+
+	// Get all app dir added or modified inside network dirs
+	const changedAppDirs = new Set();
+	for (let i = 0; i < allChangedFiles.length; i++) {
+		const dir = allChangedFiles[i].split('/').slice(0, 2).join('/');
+		if (dir.trim() && config.knownNetworks.includes(dir.split('/')[0])) {
+			changedAppDirs.add(path.resolve(dir));
+		} else {
+			throw new Error(`Added/Updated a file not belonging to network directory. filepath:${allChangedFiles[i]}`);
```
Comment (sameersubudhi) [inline validation/src/index.js:34]:
```suggestion
			throw new Error(`File (${allChangedFiles[i]}) does not belong to a known network.`);
```

File: validation/src/validateConfigFilePaths.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+const path = require('path');
+const config = require('../config');
+
+const validateConfigFilePaths = async (filePaths) => {
+	try {
+		for (let i = 0; i < filePaths.length; i++) {
+			/* eslint-disable no-await-in-loop */
+			const filePath = filePaths[i];
+			const grandparentDir = path.basename(path.dirname(path.dirname(filePath)));
```
Comment (sameersubudhi) [inline validation/src/validateConfigFilePaths.js:23]:
```suggestion
			const currentDir = path.dirname(filePath);
			const parentDir = path.dirname(currentDir);
			const grandparentDir = path.basename(parentDir);
```

File: validation/src/validateConfigFiles.js [hunk 11]
```diff
@@ -14,9 +14,8 @@
 
 const fs = require('fs').promises;
 const config = require('../config');
-const { getDirectories } = require('./utils/fs');
 
-const validateConfigInDir = async (directory) => {
+const validateAllConfigFilesForDir = async (directory) => {
```
Comment (sameersubudhi) [inline validation/src/validateConfigFiles.js:18]:
```suggestion
const validateAllConfigFilesInDir = async (directory) => {
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-validation.yml [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
```
Comment (priojeetpriyom) [inline .github/workflows/pr-validation.yml:12]: Maybe we can read node version dynamically like here?
https://github.com/LiskHQ/lisk-service/blob/feature/1157-prepare-lisk-service-v070-for-beta/.github/workflows/pr-coverage.yml#L12

**Review by priojeetpriyom — APPROVED** [review]

**Review by sameersubudhi — APPROVED** [review]
Nice job @vardan10!! 🚀 

I've created issue https://github.com/LiskHQ/app-registry/issues/24 to make some more improvements to the current implementation, so that we don't have to merge failed PRs in the future ;)

**Review by nagdahimanshu — APPROVED** [review]

**Pull request merged**